scripts/download_pages.py

```diff
@@ -349,7 +282,7 @@ def open_files():
     data_out_dir.mkdir(parents=True, exist_ok=True)
     error_file.parent.mkdir(parents=True, exist_ok=True)
 
-    thread_padding = f'{{:0{math.floor(math.log10(num_threads)) + 1}}}'
+    thread_padding = f'{{:0{num_digits(num_threads)}}}'
```

Comment:
I'm not a fan of these custom functions replacing one liners stuff.

They make the code a lot less readable. When I was reading the code for the first time, your otqdm and consume functions confused me. You are reading a code and working hard already to decipher what it does, and then you ran into an unknown function name, and you have to open another .py file to find its definition... it was quite detracting.

At least the function name num_digits is quite informative, so this is a much better case than those were. So I leave it up to you whether to go on with this num_digits function or not.